### DISCUSSION TITLE
How do I login as a bot?

### DISCUSSION THREAD

**Discussion by mahmoud-syr** [egramtel/tdsharp#184] [Q&A] [answered]
I am trying to login as a bot:
1. called await tdClient.SetTdlibParametersAsync (used test dc, provided api hash, api id, etc.)
2. Called `tdClient.CheckAuthenticationBotTokenAsync` with my Bot token

Getting `TdLib.TdException: 'ACCESS_TOKEN_INVALID'`

Would have been nice if there is a documentation for the library as the methods aren't really self-explanatory nor the error messages.

Thanks in advance

**Comment by ForNeVeR** [comment]
Sorry, no idea on that. You have to ask the upstream maintainers about that. All the available documentation is included in the XML doc.

**Reply by mahmoud-syr** [reply]
Thanks for your answer. Would you be able to guide me as to where exactly can I start asking question? Who are these upstream maintainers, you mean in the `tdlib` repository?
Thanks!

**Reply by mahmoud-syr** [reply]
Also the method name here is `checkAuthenticationBotToken` but in the documentation is said it should be [auth.importBotAuthorization](https://core.telegram.org/method/auth.importBotAuthorization)

**Reply by ForNeVeR** [reply, marked as answer]
I mean this repository: https://github.com/tdlib/td

Regarding "auth.importBotAuthorization", I don't understand this notation. We import data from the `tl` files published by Telegram in the upstream repo, and apparently there's a method called `checkAuthenticationBotToken`: https://github.com/egramtel/tdsharp/blob/9dcb8ea4965b4049a304e4e0b2eff07f5cc2008a/TdLib.CodeGen/Methods.tl#L76-L77

**Reply by mahmoud-syr** [reply]
This is so confusing. The official documentation mention the `importBotAuthorization` but then there is only this method which for some reason is not accepting my bot token.
I'll create an issue there then.
Thank you very much for your help.
